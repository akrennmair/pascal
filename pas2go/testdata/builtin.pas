program test;

var i, j : integer;
	r, s : real;

begin
	i := -23;
	j := abs(i);
	r := -3.1415;
	s := abs(r);
	writeln('abs(', i, ') = ', j);
	writeln('abs(', r, ') = ', s);

	s := arctan(r);
	writeln('arctan(', r, ') = ', s)
end.
