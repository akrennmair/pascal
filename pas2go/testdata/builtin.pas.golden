package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	var (
		i int
		j int
		r float64
		s float64
	)
	_ = i
	_ = j
	_ = r
	_ = s

	i = (-23)
	j = system.AbsInt(i)
	r = (-3.1415e0)
	s = system.AbsReal(r)
	system.Writeln("abs(", i, ") = ", j)
	system.Writeln("abs(", r, ") = ", s)
	s = system.Arctan(r)
	system.Writeln("arctan(", r, ") = ", s)
}
